File: JavaProgLang/BinaryToDec.java
import java.util.*;
public class BinaryToDec {
    public static int BinToDec( int binNum){
        int pow=0,dec=0,lastDigit,DecNum=0;
        while(binNum>0){
            lastDigit=binNum%10;
            DecNum=lastDigit+(lastDigit*(int)Math.pow(2,pow));
            pow++;
            binNum/=10;
        }
        return DecNum;
    }
    public static int DectoBin(int n){
        int pow=0,rem,BinaryNum=0;
        while(n>0){
            rem=n%2;
            BinaryNum=BinaryNum+(rem*(int)Math.pow(10,pow));
            pow++;
            n/=2;
        }
        return BinaryNum;
    }
    public static void main(String args[]){
        // binary to decimal
        System.out.println("Enter a binary number");
        Scanner s=new Scanner(System.in);
        int binNum=s.nextInt();
        System.out.println("Binary equivalent of "+binNum+" is :"+BinToDec(binNum));


        //Decimal to binary
        System.out.println("Enter a Decimal number ");
        int Decum=s.nextInt();
        System.out.println("Decimal equivalent of "+Decum+" is : "+DectoBin(Decum));
    }
}

File: C/BinaryTodec.c
#include<stdio.h>
#include<Math.h>
int BintoDec(int n){
    int power=0,dec=0,lastDigit;
    while(n>0){
        lastDigit=n%10;
        dec=lastDigit+(lastDigit*(int)pow(2,power));
        power++;
        n/=10;
    }
    return dec;
}
int DectoBin(int n){
    int power=1,rem,BinaryNum=0;
    while(n>0){
        rem=n%2;
        BinaryNum=BinaryNum+(rem*power);
        power*=10;
        n/=2;
    }
    return BinaryNum;
}
int main(){
    printf("\nEnter the binary number");
    int BinNum;
    scanf("%d",&BinNum);
    printf("Decimal equivalent of %d is %d\n", BinNum, BintoDec(BinNum));

    //decimal to binary
    printf("\nEnter the Decimal number");
    int Decnum;
    scanf("%d",&Decnum);
    printf("Binary equivalent of %d is %d\n", Decnum, DectoBin(Decnum));

}